### DISCUSSION TITLE
CodeBlockLowlight for svelte or some way to pass custom component to this extension?

### DISCUSSION THREAD

**Discussion by dartilesm** [ueberdosis/tiptap#1392] [General Questions] [answered]
Hi everyone!

I'm trying to do something like [this](https://www.tiptap.dev/examples/syntax-highlighting) with svelte. When i use the `addNodeView` method inside extend, i get `addNodeView(...) is not a function` error message.

I did make a sample in codesanbox https://codesandbox.io/s/svelte-codeblocklowlight-tiptap-7s1ir?file=/Editor.svelte

**How to reproduce it** 
When select a text and click on the code icon, we get the error that I mention.

Any ideas?

Thanks a lot!

**Comment by philippkuehn** [comment, marked as answer]
You can’t return `null`. A method is expected. https://www.tiptap.dev/guide/node-views/js#render-a-node-view-with-javascript

**Reply by dartilesm** [reply]
I returned null, just to return something but before asking this question I tried many things including this https://www.tiptap.dev/guide/node-views/js#render-a-node-view-with-javascript

Now that I try again it works for me and the button with the label "Node view" appears.
I think the "rubber duck debugging" happened to me

This It wasn't exactly what I needed, but it will help me move forward.  🤩

I really like tiptap, I hope to become a sponsor soon 

Thanks a lot!!
